### ISSUE TITLE
Outer product of QN ITensors crashing

### ISSUE THREAD

**Issue by emstoudenmire** [ITensor/ITensors.jl#208]
I am seeing a crash, happening in Tensors/contraction_logic.jl line 306, when contracting together two QNITensors. These two ITensors have no indices in common, which may make it a corner case that has a bug.

Here is a minimal example to reproduce the bug:

```
  l0 = Index(QN()=>2;tags="Link,n=0")
  l1 = Index(QN()=>2,QN(("Sz",-2)) => 1,QN(("Sz",2)) => 1;tags="Link,n=1")
  IT = ITensor(l0,l1)
  IT[l0(2),l1(4)] = 0.5
  @show IT

  s1 = Index(QN(("Sz",1)) => 1,QN(("Sz",-1)) => 1;tags="S=1/2,Site,n=1")
  Op = ITensor(s1',dag(s1))
  Op[s1'(1),dag(s1)(2)] = 1.0
  @show Op

  R = IT * Op
  @show R
```

**Comment by mtfishman** [comment]
Yes, I noticed this. It is fixed in #206.

**Comment by emstoudenmire** [comment]
Great! PR #206 also fixed the issue I was having on the QN AutoMPO branch I’m working on, after I merged in the latest master branch.
